### PR TITLE
[sival/entropy_src] Add entropy_src_bypass_mode_health_test

### PR DESCRIPTION
**Pull request by github-gcontributor**
This test implements the entropy source test,
"chip_sw_entropy_src_bypass_mode_health_tests" scenario as described in the OpenTitan testplan.

- ENTROPY_SRC provides one initial seed in boot-time / bypass mode.
- Signals a recoverable alert if the total failure counter hits the threshold in boot-time / bypass mode.
- Checks and prints status registers for debug

See: /hw/top_earlgrey/data/ip/chip_entropy_src_testplan.hjson

### CODE REVIEW

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (nasahlpa) [inline sw/device/tests/entropy_src_bypass_mode_health_test.c:152]: What is the expected result here? We should add a check whether we get the expected result.
Same applies throughout the test.

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (nasahlpa) [inline sw/device/tests/entropy_src_bypass_mode_health_test.c:263]: Cycles is a bit misleading here as it is actually an iteration counter.

**Review by nasahlpa — COMMENTED** [review]
Thanks for your work, @github-gcontributor! I've added a couple of suggestions.

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (github-gcontributor) [reply sw/device/tests/entropy_src_bypass_mode_health_test.c:306]: As per my understanding, 
return OK_STATUS() means "everything succeeded."
return INTERNAL() means "something failed."
Do we have any other better way to showcase error?

I do see ./sw/device/lib/base/status.h has "FAILED_PRECONDITION"

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (github-gcontributor) [reply sw/device/tests/entropy_src_bypass_mode_health_test.c:148]: Noted!

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (github-gcontributor) [reply sw/device/tests/entropy_src_bypass_mode_health_test.c:126]: Noted!

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (github-gcontributor) [reply sw/device/tests/entropy_src_bypass_mode_health_test.c:180]: Noted!

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (github-gcontributor) [reply sw/device/tests/entropy_src_bypass_mode_health_test.c:142]: These were used for debugging if there hang is due to some other cases
I shall remove them

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (github-gcontributor) [reply sw/device/tests/entropy_src_bypass_mode_health_test.c]: Noted!

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (github-gcontributor) [reply sw/device/tests/entropy_src_bypass_mode_health_test.c:252]: Sure, I shall have better variable names like kIterateMaxRetries and iter_cntr

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (nasahlpa) [reply sw/device/tests/entropy_src_bypass_mode_health_test.c:287]: Looks good, thanks.

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (nasahlpa) [inline sw/device/tests/entropy_src_bypass_mode_health_test.c:399]: You also could replace this check with `TRY_CHECK()`.

File: sw/device/tests/BUILD
Comment (nasahlpa) [inline sw/device/tests/BUILD:1479]: Do we need to map this test to a SiVal test?

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (github-gcontributor) [reply sw/device/tests/entropy_src_bypass_mode_health_test.c:68]: Noted!

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (github-gcontributor) [reply sw/device/tests/entropy_src_bypass_mode_health_test.c:180]: Noted!

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (github-gcontributor) [reply sw/device/tests/entropy_src_bypass_mode_health_test.c:378]: Noted!

File: sw/device/tests/entropy_src_bypass_mode_health_test.c
Comment (github-gcontributor) [reply sw/device/tests/entropy_src_bypass_mode_health_test.c:399]: Noted!

File: sw/device/tests/BUILD
Comment (github-gcontributor) [reply sw/device/tests/BUILD:1479]: I am not sure. @engdoreis if you can comment on this?

**Review by nasahlpa — APPROVED** [review]
Thanks, LGTM!
But I'd suggest waiting for @engdoreis answer regarding SiVal test before merging

File: sw/device/tests/BUILD
Comment (engdoreis) [reply sw/device/tests/BUILD:1479]: yes, you have to link this test in the `bazel_target` field of the testplan, otherwise it will now reflect in the A1 Sheet.

**Pull request merged**
